Use python zipfile to have x-platform idempotent lambda packages

Felixoid opened the pull request:
<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix issue with building lambdas on linux and osx.

Felixoid committed:
Use python zipfile to have x-platform idempotent lambda packages

robot-ch-test-poll commented:
<!-- automatic status comment for PR #60603 from ClickHouse/ClickHouse:zip-lambda-in-docker -->
*This is an automated comment for commit 0aaef90097dc09f6afe655567fd11000d16e8149 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60603/0aaef90097dc09f6afe655567fd11000d16e8149/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

Felixoid committed:
Remove python bytecode, make consistent file-permissions

It's impossible to have persistent pyc files
Each time they are built with different content, for example:

> cmp -bl lambda-package*/charset_normalizer/__pycache__/constant.cpython-310.pyc
15582   6 ^F     4 ^D
15583 164 t    155 m
15584 141 a    142 b
15586 164 t    163 s
15587 151 i    332 M-Z
15588 163 s      6 ^F
15589 332 M-Z  164 t
15590   4 ^D   141 a
15591 155 m    143 c
15592 142 b    164 t
15593 143 c    151 i
17425  74 <     75 =
17428  76 >     46 &
17429 332 M-Z  372 M-z
17431 173 {     55 -
17434  75 =    174 |
17437  57 /     72 :
17440  54 ,     73 ;
17441 372 M-z  332 M-Z
17443 174 |    175 }
17446  55 -     54 ,
17447 372 M-z  332 M-Z
17449  46 &    173 {
17452  72 :     76 >
17455  42 "     74 <
17458  73 ;    133 [
17461 135 ]     42 "
17464 133 [    135 ]
17465 332 M-Z  372 M-z
17467 175 }     57 /
17503 332 M-Z  162 r
17504   5 ^E   130 X
17505 152 j      0 ^@
17506 157 o      0 ^@
17507 150 h      0 ^@
17508 141 a    332 M-Z
17509 142 b      5 ^E
17510 162 r    152 j
17511 130 X    157 o
17512   0 ^@   150 h
17513   0 ^@   141 a
17514   0 ^@   142 b
17536   5 ^E     2 ^B
17537 143 c    150 h
17538 160 p    172 z
17539  71 9    332 M-Z
17540  65 5      5 ^E
17541  60 0    143 c
17542 332 M-Z  160 p
17543   2 ^B    71 9
17544 150 h     65 5
17545 172 z     60 0
